com.unity.shadergraph/Editor/ShaderFoundry/Linker/LegacyTemplateLinker.cs

```diff
@@ -563,6 +576,11 @@ void ProcessBlockInstance(BlockInstance blockInstance, VisitedRegistry visitedRe
                     shaderCommands.AddRange(block.Commands);
                     shaderDefines.AddRange(block.Defines);
                     shaderIncludes.AddRange(block.Includes);
+
+                    // TODO: there's probably a better way to gather includes from all functions -- like put this gathering in the visitor
```

Comment:
Yeah, ideally this would be done it the traversal that finds all used functions/types and declares them in the correct order, however this is fine as the order is mostly independent.